specs/gloas/beacon-chain.md

```diff
@@ -1239,7 +1540,10 @@ def process_execution_payload(
 
     # Verify signature
     if verify:
-        assert verify_execution_payload_envelope_signature(state, signed_envelope)
+        if signed_envelope.message.builder_index == BUILDER_INDEX_SELF_BUILD:
+            assert signed_envelope.signature == bls.G2_POINT_AT_INFINITY
```

Comment:
hmm, I don't think this is safe, this allows anyone to send an equivocation envelope, we should case the self-building case and actually verify against the proposer signature .